pkg/download/mode/mode.go

```diff
@@ -44,17 +58,30 @@ type DownloadPath struct {
 	DownloadURL string `hcl:"downloadURL,optional"`
 }
 
+func (d DownloadPath) validate(op errors.Op) error {
+	if d.DownloadURL == "" && (d.Mode == Redirect || d.Mode == AsyncRedirect) {
+		return errors.Config(
+			op,
+			fmt.Sprintf("DownloadURL (inside %s in the download file)", d.Pattern),
+			"You must set a value when the download mode is 'redirect' or 'async_redirect'",
+			"https://docs.gomods.io/configuration/download/",
+		)
+	}
+	return nil
+}
+
 // NewFile takes a mode and returns a DownloadFile.
 // Mode can be one of the constants declared above
 // or a custom HCL file. To pass a custom HCL file,
 // you can either point to a file path by passing
 // file:/path/to/file OR custom:<base64-encoded-hcl>
 // directly.
 func NewFile(m Mode, downloadURL string) (*DownloadFile, error) {
+	const downloadModeURL = "https://docs.gomods.io/configuration/download/"
```

Comment:
I would make this a package level constant (private) as it is being used 3 more times in the pr